FuncGodot Manual/pages/tips_runtime_building.html

```diff
@@ -0,0 +1,49 @@
+        <p>When the map builds, it will emit the signals <code>build_complete</code> or <code>build_failed</code>, which you can then connect to, to handle errors gracefully and to run any post-build code you need to run.</p>
+        <p>If you need the UV2 information for baking lightmaps, you can then call FuncGodotMap's <code>unwrap_uv2()</code> method. This generates the UV2 information, which emits the signal <code>unwrap_uv2_complete</code> when it is complete. It is at this point you may bake your Lightmaps.
+        <blockquote><i>As of Godot 4.2, runtime lightmap baking is not currently possible, so instead you should consider using a different lighting solution here.</i></blockquote>
+        An example of runtime map building is as follows:</p>
+        <pre><code>extends Node
+
+@onready var func_godot_map := $FuncGodotMap as FuncGodotMap
+
+# We can connect the various build signals of the FuncGodotMap 
+# in order to add more functionality after building.
+func _ready():
+    func_godot_map.connect("build_complete", _build_complete)
+    func_godot_map.connect("build_failed", _build_failed)
+    func_godot_map.connect("unwrap_uv2_complete", _unwrap_uv2_complete)
+    func_godot_map.verify_and_build()
+
+# We need to wait for the build to finish before unwrapping mesh UV2s for lightmap baking.
+# Currently lightmap baking cannot be performed at runtime, so it's not terribly useful yet.
+# Consider creating a procedural Voxel GI solid class entity that can be baked at runtime instead.
+func _build_complete() -> void:
+    print("Success! Unwrapping UV2...")
+    func_godot_map.unwrap_uv2()
+
+func _build_failed() -> void:
+    printerr("Failed to build the map file! :(")
+
+func _unwrap_uv2_complete() -> void:
+    print("UV2 unwrapping completed!")</code></pre>
+    <p>An example of this can be found in <a href="https://github.com/func-godot/func_godot_runtime_building" target="_blank">func_godot_runtime_building repository</a>.
+    <blockquote><i>NOTE: This approach has limitations. You will need to figure out solutions to anything that needs baking, or for any scene changes that need to be made after the map is built.</i></blockquote>
+    <blockquote><i>NOTE: Runtime map building runs into issues where .map files are saved as a .tres in your exported game. To get around this, you can change your import type to Skip and add <code>*map</code> to your non-resource export filter. Since 4.3 you can change the import type to Keep.</i></blockquote>
+    </p>
+    <h2>Using command lines to build from Map Editors</h2>
+    <p>Combined with <code>OS.get_cmdline_user_args();</code> and your map editor's build profile, you can open up a map you are editing and build it at runtime to test it. For Trenchbroom, this involves <a href="https://trenchbroom.github.io/manual/latest/#launching_game_engines">configuring launch options</a>.</p>
```

Comment:
I think I'll wait for @sinewavey to add her part before I go all Jay Sherman on it.

Reply:
See [here](https://github.com/func-godot/func_godot_docs/pull/10) - ready for feedback whenever. I'd hoped to get some other thoughts on it before finalizing it

Reply:
I've added this in. I've expanded the Trenchbroom section slightly with of the short of what is needed(configuring game engine, and passing parameters). I've also added a note about executables, since it may not be obvious you can run the editor with a path to the project as the executable

Reply:
Looking through the new images showing the map editor settings, it'd be best to not use images or examples from our personal projects, but rather from a bare bones basic project like how the runtime example project is set up, for both clarity and consistency. I know it's easier to just use your pre-existing projects, but those can also come with pre-existing biases to personal work flows rather than trying to find the simplest path forward, and to better understand how to duplicate the processes for these features.

If at all possible, I would recommend avoiding personal user folders for example project paths as well (similar to how all of the other examples in the manual are effectively `C:\GAME DEV\FuncGodot\` or some similar variation).

Reply:
The trenchbroom section has additional screenshots, though I'm limited in this for making obvious file paths since I use Linux and don't have any Windows systems currently. Also means the UI is GTK-based(iirc) so is themed, unlike default TrenchBroom on windows.

I would love for someone else to make screenshots that are more accessible than the ones I currently have

Reply:
Here are some generic Func_Godot referencing screenshots:

![image](https://github.com/func-godot/func_godot_docs/assets/90056029/66bb3597-2156-4ae9-821c-b9056c85f080)
![image](https://github.com/func-godot/func_godot_docs/assets/90056029/7a7db3d1-99b8-4377-ad62-b81217984d6a)
![image](https://github.com/func-godot/func_godot_docs/assets/90056029/fe35fa00-45f8-4afd-9c03-fe41993314fb)

Reply:
I've addressed this in the latest commit. Let me know if there's any further thoughts on these